model/common/src/icon4py/model/common/grid/grid_manager.py

```diff
@@ -496,27 +519,30 @@ def _read_start_end_indices(
         return start_indices, end_indices, grid_refinement_dimensions
 
     def _read_grid_refinement_fields(
-        self, decomposition_info: Optional[decomposition.DecompositionInfo] = None
-    ) -> tuple[dict[dims.Dimension : xp.ndarray]]:
+        self,
+        backend: gtx_backend.Backend,
+        decomposition_info: Optional[decomposition.DecompositionInfo] = None,
+    ) -> tuple[dict[dims.Dimension : NDArray]]:
         """
         Reads the refinement control fields from the grid file.
 
         Refinement control contains the classification of each entry in a field to predefined horizontal grid zones as for example the distance to the boundaries,
         see [refinement.py](refinement.py)
         """
+        np = common_utils.gt4py_field_allocation.import_array_ns(backend)
```

Comment:
since we use `np` for numpy direct import, would t be better to rename this to something else?

Reply:
Right, should be `xp` really.